include/boost/numeric/odeint/stepper/adaptive_adams_bashforth.hpp

```diff
@@ -0,0 +1,152 @@
+#define STEPPER_ADAMS_BASHFORTH_HPP_INCLUDED
+
+#include <boost/numeric/odeint/stepper/detail/polynomial.hpp>
+#include <boost/numeric/odeint/stepper/detail/adaptive_adams_coefficients.hpp>
+
+#include <boost/numeric/odeint/stepper/base/algebra_stepper_base.hpp>
+#include <boost/numeric/odeint/algebra/algebra_dispatcher.hpp>
+#include <boost/numeric/odeint/algebra/operations_dispatcher.hpp>
+
+#include <boost/numeric/odeint/stepper/stepper_categories.hpp>
+
+#include <boost/numeric/odeint/util/copy.hpp>
+#include <boost/numeric/odeint/util/bind.hpp>
+
+#include <boost/numeric/odeint/util/state_wrapper.hpp>
+#include <boost/numeric/odeint/util/is_resizeable.hpp>
+#include <boost/numeric/odeint/util/resizer.hpp>
+
+#include <iostream>
+
+namespace boost {
+namespace numeric {
+namespace odeint {
+
+template<
+size_t Steps,
+class State,
+class Value = double,
+class Deriv = State,
+class Time = Value,
+class Algebra = typename algebra_dispatcher< State >::algebra_type,
+class Operations = typename operations_dispatcher< State >::operations_type ,
+class Resizer = initially_resizer
+>
+class adaptive_adams_bashforth: public algebra_stepper_base< Algebra , Operations >
+{
+	public:
+		static const size_t steps = Steps;
+		typedef unsigned short order_type;
+		static const order_type order_value = steps;
+
+		typedef State state_type;
+		typedef Value value_type;
+		typedef Deriv deriv_type;
+		typedef Time time_type;
+		typedef Resizer resizer_type;
+
+		typedef Algebra algebra_type;
+		typedef Operations operations_type;
+		typedef algebra_stepper_base< Algebra , Operations > algebra_stepper_base_type;
+
+		typedef state_wrapper<state_type> wrapped_state_type;
+		typedef state_wrapper<deriv_type> wrapped_deriv_type;
+		typedef stepper_tag stepper_category;
+
+		typedef detail::adaptive_adams_coefficients<Steps, deriv_type, time_type, algebra_type, operations_type> coeff_type;
+
+		typedef adaptive_adams_bashforth< Steps , State , Value , Deriv , Time , Algebra, Operations, Resizer > stepper_type;
+
+		adaptive_adams_bashforth( const algebra_type &algebra = algebra_type() )
+		:algebra_stepper_base_type( algebra ) ,
+		m_dxdt_resizer(), m_xnew_resizer()
+		{};
+
+		order_type order() const
+		{
+			return order_value;
+		};
+
+		template<class System>
+		void do_step(System system, state_type & inOut, time_type t, time_type dt)
+		{
+			m_xnew_resizer.adjust_size( inOut , detail::bind( &stepper_type::template resize_xnew_impl< state_type > , detail::ref( *this ) , detail::_1 ) );
+
+			do_step(system, inOut, t, m_xnew.m_v, dt);
+			boost::numeric::odeint::copy( m_xnew.m_v , inOut);
+		};
+
+		template<class System>
+		void do_step(System system, const state_type & in, time_type t, state_type & out, time_type dt)
+		{
+			m_dxdt_resizer.adjust_size( in , detail::bind( &stepper_type::template resize_dxdt_impl< state_type > , detail::ref( *this ) , detail::_1 ) );
+
+			system(in, m_dxdt.m_v, t);
+			m_coeff.step(m_dxdt.m_v, t);
+			m_coeff.confirm();
```

Comment:
I don't think there is any reason to use adaptive_adams_bashforth over adams_bashforth. But in that case, it would probably make more sense to remove adaptive_adams_bashforth and adaptive_adams_moulton altogether and move the complete logic into adaptive_adams_bashforth_moulton.